src/CellData/DiracDeltas.jl

```diff
@@ -65,14 +70,63 @@ function DiracDelta{0}(model::DiscreteModel;tags)
   DiracDelta{0}(model,degree;tags=tags)
 end
 
-function (d::DiracDelta)(f)
+function (d::GenericDiracDelta)(f)
   evaluate(d,f)
 end
 
-function evaluate!(cache,d::DiracDelta,f)
+function evaluate!(cache,d::GenericDiracDelta,f)
   ∫(f)*d.dΓ
 end
 
-function get_triangulation(d::DiracDelta)
+function get_triangulation(d::GenericDiracDelta)
   d.Γ
 end
+
+# For handling DiracDelta at a generic Point in the domain #
+
+function _cell_to_pindices(pvec::Vector{<:Point},trian::Triangulation)
+  cache = _point_to_cell_cache(KDTreeSearch(),trian)
+  cell_to_pindex = Dict{Int, Vector{Int32}}()
+  for i in 1:length(pvec)
+    cell = _point_to_cell!(cache, pvec[i])
+    push!(get!(() -> valtype(cell_to_pindex)[], cell_to_pindex, cell), i)
+  end
+  cell_to_pindex
+end
+
+function DiracDelta(model::DiscreteModel{D}, p::Point{D,T}) where {D,T}
+  trian = Triangulation(model)
+  cache = _point_to_cell_cache(KDTreeSearch(),trian)
+  cell = _point_to_cell!(cache, p)
+  trianv = TriangulationView(trian,[cell])
+  pquad = GenericQuadrature([p],[one(T)])
+  pmeas = Measure(CellQuadrature([pquad],[[p]],[[one(T)]],trianv,PhysicalDomain(),PhysicalDomain()))
+  GenericDiracDelta{0,D,NotGridEntity}(trianv,pmeas)
+end
+
+function DiracDelta(model::DiscreteModel{D}, pvec::Vector{Point{D,T}}) where {D,T}
+  trian = Triangulation(model)
+  cell_to_pindices = _cell_to_pindices(pvec,trian)
+  cell_ids = collect(keys(cell_to_pindices))
+  cell_points = collect(values(cell_to_pindices))
+  points = map(i->pvec[cell_points[i]], 1:length(cell_ids))
+  weights_x_cell = collect.(Fill.(one(T),length.(cell_points)))
+  pquad = map(i -> GenericQuadrature(pvec[cell_points[i]],weights_x_cell[i]), 1:length(cell_ids))
+  trianv = Triangulation(trian,cell_ids)
+  pmeas = Measure(CellQuadrature(pquad,points,weights_x_cell,trianv,PhysicalDomain(),PhysicalDomain()))
+  GenericDiracDelta{0,D,NotGridEntity}(trianv,pmeas)
+end
+
+function evaluate!(cache,d::GenericDiracDelta{0,Dt,NotGridEntity},f::Function) where Dt
```

Comment:
Which is the ultimate reason behind having to define this specialised method? 

Why cant we leverage the generic implementation of `evaluate!` whenever the domain of `DiracDelta` is a function?

Reply:
I was in a misconception that existing `evaluate!` method goes through the search process of Point to Cell (which ofcourse works perfectly fine as well), which is only necessary for `CellFields`, Functions can be evaluated directly without the search. But it seems it doesn't go through that route once we have the `CellPoint` available, which is very nice. So I ll remove it..

Reply:
But one advantage of the new constructor is it is ~4 times faster and aloocates ~10 times less memory.

Reply:
> But it seems it doesn't go through that route once we have the CellPoint available, which is very nice.

Whenever you have a `CellPoint`, Gridap assumes that the points in each cell are within the cell (either reference cell or physical cells, depending on the value of the corresponding trait). 

If the points are not within the cell, then I guess you have an undefined behaviour. But we are satisfying this precondition, as the cells have been judiciously selected in the TriangulationView such that the hold the corresponding points.

Reply:
> But one advantage of the new constructor is it is ~4 times faster and aloocates ~10 times less memory.

Do these ratios keep asympotatically fixed as you refine the mesh?

Reply:
> Whenever you have a CellPoint, Gridap assumes that the points in each cell are within the cell (either reference cell or physical cells, depending on the value of the corresponding trait).

>If the points are not within the cell, then I guess you have an undefined behaviour. But we are satisfying this precondition, as the cells have been judiciously selected in the TriangulationView such that they hold the corresponding points.

Yes true.. since we are checking it, it should be safe, it is just a matter of efficiency

Reply:
> Do these ratios keep asympotatically fixed as you refine the mesh?
I ll check and update.

Reply:
On refinement the the new version runs in the same time but the earlier one becomes slower and more memory intensive.